flows/knack/knack_services.py

```diff
@@ -0,0 +1,400 @@
+docker_env = "production"
+docker_image = f"atddocker/atd-knack-services:{docker_env}"
+
+# Set flow runtime name from:
+# https://github.com/PrefectHQ/prefect/discussions/3881
+def set_run_name(flow, old_state, new_state):
+    if new_state.is_running():
+        client = prefect.Client()
+        name = f"{prefect.context.parameters['App Name']}:{prefect.context.parameters['Knack Container']}-{prefect.context.date}"  # use flow-name-day-of-week as the flow run name, for example
+        client.set_flow_run_name(prefect.context.flow_run_id, name)
+
+
+# Based on inputs, determine if some conditional tasks should run
+@task(name="determine_task_runs", nout=2)
+def determine_task_runs(layer, app_name_dest):
+    return bool(layer), bool(app_name_dest)
+
+
+# Task to pull the latest Docker image
+@task(
+    name="pull_docker_image",
+    max_retries=1,
+    timeout=timedelta(minutes=60),
+    retry_delay=timedelta(minutes=5),
+    # state_handlers=[handler],
+    log_stdout=True,
+)
+def pull_docker_image():
+    client = docker.from_env()
+    response = client.images.pull("atddocker/atd-knack-services", all_tags=True)
+    logger.info(docker_env)
+    return response
+
+
+# Get the envrioment variables for the given app
+@task(
+    name="get_env_vars",
+    task_run_name="get_env_vars:{app}",
+    max_retries=1,
+    timeout=timedelta(minutes=60),
+    retry_delay=timedelta(minutes=5),
+    # state_handlers=[handler],
+    log_stdout=True,
+)
+def get_env_vars(app):
+    environment_variables = get_key_value(key=f"{app}-atd_knack_services")
+    return environment_variables
+
+
+# Get the last date (string) the flow was succesful
+@task(
+    name="get_last_exec_time",
+    task_run_name="get_last_exec_time (app: {app}, container: {container})",
+    max_retries=1,
+    timeout=timedelta(minutes=60),
+    retry_delay=timedelta(minutes=5),
+    # state_handlers=[handler],
+    log_stdout=True,
+)
+def get_last_exec_time(app, container, replace_data):
+    # Completely replace data on 15th day of every month,
+    # to catch records potentially missed by incremental refreshes
+    if datetime.today().day == 15:
+        return "1970-01-01"
+
+    # parameter option to replace all data
+    if replace_data:
+        return "1970-01-01"
+
+    # Get dict of previous executions
+    prev_execs = get_key_value("atd_knack_services_prev_exec")
+
+    # Key is unique based on the container and app
+    key = f"{app}:{container}-prev-exec"
+
+    # Return the date for this container/app if it exists
+    if key in prev_execs:
+        return prev_execs[key]
+    else:
+        # Return this if it doesn't exist yet
+        return "1970-01-01"
+
+
+# Records to postgrest
+@task(
+    name="records_to_postgrest",
+    task_run_name="records_to_postgrest (app: {app_name}, contianer: {container}, date_filter:{date_filter})",
+    max_retries=1,
+    timeout=timedelta(minutes=60),
+    retry_delay=timedelta(minutes=5),
+    # state_handlers=[handler],
```

Comment:
I see this line commented out. Is this something you can maybe remove everywhere before you merge, or comment back in?

Reply:
Thank you, I'll zap these #'s to make sure we get slack notifications